File: merging/js/bubblesort.js
function bubbleSort(arr) {
  const length = arr.length;
  for (let i = 0; i < length - 1; i++) {
    let flag = 0;
    for (let j = 0; j < length - 1 - i; j++) {
      if (arr[j] > arr[j + 1]) {
        let temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        flag = 1;
      }
    }
    if (flag === 0) break;
  }
  return arr;
}

// console.log(bubbleSort([10, 3, 9, 1, 7, 6, 8, 0, 4, 5, 2]));
console.log(bubbleSort([2, 9, 1]));

File: merging/python/bubblesort.py

# we make bubble sort adaptive by using a flag 
# adaptive means it takes the least amount of time if the list is already sorted
# bubble sort is stable
def bubble_sort(list):
  length = len(list)
  for i in range((length)):
    flag = 0
    for j in range((length - 1 - i)):
      if list[j] > list[j + 1]:
        temp = list[j]
        list[j] = list[j + 1]
        list[j + 1] = temp
        flag = 1
    if flag == 0:
      break
  return list




   
    


print(bubble_sort([4,2,7]))